restapi/src/main/java/io/stargate/web/docsapi/resources/CollectionsResource.java

```diff
@@ -0,0 +1,232 @@
+          @HeaderParam("X-Cassandra-Token")
+          String token,
+      @ApiParam(value = "the namespace to create the collection in", required = true)
+          @PathParam("namespace-id")
+          String namespace,
+      @ApiParam(
+              value = "JSON with the name of the collection",
+              required = true,
+              example = "{\"name\": \"example\"}")
+          String payload) {
+    return RequestHandler.handle(
+        () -> {
+          DocumentDB docDB = db.getDocDataStoreForToken(token);
+          DocCollection info = mapper.readValue(payload, DocCollection.class);
+          if (info.getName() == null) {
+            throw new IllegalArgumentException("`name` is required to create a collection");
+          }
+          boolean res =
+              collectionService.createCollection(namespace, info.getName(), docDB, db.isDse());
+          if (res) {
+            return Response.status(Response.Status.CREATED).build();
+          } else {
+            return Response.status(Response.Status.CONFLICT)
+                .entity(
+                    String.format("Create failed: collection %s already exists", info.getName()))
+                .build();
+          }
+        });
+  }
+
+  @DELETE
+  @ApiOperation(value = "Delete a collection in a namespace")
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 204, message = "No Content"),
+        @ApiResponse(code = 404, message = "Not Found"),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 500, message = "Internal server error", response = Error.class)
+      })
+  @Path("collections/{collection-id}")
+  @Consumes("application/json")
+  @Produces("application/json")
+  public Response deleteCollection(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String token,
+      @ApiParam(value = "the namespace containing the collection to delete", required = true)
+          @PathParam("namespace-id")
+          String namespace,
+      @ApiParam(value = "the collection to delete", required = true) @PathParam("collection-id")
+          String collection) {
+    return RequestHandler.handle(
+        () -> {
+          DataStore localDB = db.getDataStoreForToken(token);
+          Table toDelete = localDB.schema().keyspace(namespace).table(collection);
+          if (toDelete == null) {
+            return Response.status(Response.Status.NOT_FOUND)
+                .entity(String.format("Collection %s not found", collection))
+                .build();
+          }
+          collectionService.deleteCollection(namespace, collection, new DocumentDB(localDB));
+          return Response.status(Response.Status.NO_CONTENT).build();
+        });
+  }
+
+  @POST
+  @ApiOperation(value = "Upgrade a collection in a namespace")
```

Comment:
This probably needs a warning about what could happen with availability during the upgrade.

Reply:
Yea I will add more docs here